woocommerce/woocommerce
# Migrate select control component to TS

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes #35380.

This PR migrates `@woocommerce/components/select-control` to TS



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter=@woocommerce/storybook run storybook`
2. Go to http://localhost:6007/?path=/story/woocommerce-admin-components-selectcontrol--basic
3. Test all storybook examples
4. Ensure that select controls work as expected
5. Go to the WooCommerce onboarding wizard.
- You can do this by installing and activating WooCommerce in a brand new site.
- Or, via the **WooCommerce ▸ Settings screen**, locate the Help pulldown and re-run the wizard that way:

![WooCommerce ▸ Settings screen > OBW](https://user-images.githubusercontent.com/3594411/207470607-dfd82ba7-de97-43c2-b691-d76c0eabc2d1.png)
6. Confirm you can select a country.
